wormhole-connect/src/hooks/useResumeTransaction.ts

```diff
@@ -0,0 +1,79 @@
+import { useContext, useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import config, { getWormholeContextV2 } from 'config';
+import { RouteContext } from 'contexts/RouteContext';
+import {
+  setRoute as setRedeemRoute,
+  setIsResumeTx,
+  setTimestamp,
+  setTxDetails,
+} from 'store/redeem';
+import { setRoute as setAppRoute } from 'store/router';
+import { setToChain } from 'store/transferInput';
+
+import { TransactionLocal } from 'config/types';
+
+const useResumeTransaction = (
+  props: TransactionLocal | undefined,
+): { isLoading: boolean; error?: string } => {
+  const dispatch = useDispatch();
+
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const routeContext = useContext(RouteContext);
+
+  const { receipt, route, timestamp, txDetails, txHash } = props || {};
+
+  useEffect(() => {
```

Comment:
I was confused reading this code at first because IMO it's a somewhat unconventional way to use a React effect. You're just waiting for the `props` to not equal `undefined` a single time, and then running an async callback function.

It seems like this could just be a single `async` utility function, which `Widget/Item.tsx` imports & calls. The loading state could be handled locally by `Item` and the error could just be thrown.

What do you think? Probably less code, and simpler code, to do it that way.

Reply:
First of all the "props" dep in the side-effect is wrong, it should be these four props: receipt, route, timestamp, txDetails. I'll fix that.
I initially added this inside the widget/item as a local function, but I wanted to make it sharable so can be used by TxSearch at some point. The reason I wanted to make it a hook is to separate this behavior from components and handle everything in hook's local state. But the functionality seems much less complicated than I previously anticipated. Let me move this into a utility function and see if that makes more sense 👍

Reply:
sounds good. yeah we can make it reusable, just as a utility function. that's essentially all it is.